replica/table.cc

```diff
@@ -791,26 +791,33 @@ table::start() {
 future<>
 table::stop() {
     if (_async_gate.is_closed()) {
-        return make_ready_future<>();
+        co_return;
+    }
+    std::exception_ptr ex;
+    try {
+        co_await _async_gate.close();
+        co_await await_pending_ops();
+    } catch (...) {
+        ex = std::current_exception();
+    }
+    try {
+        co_await _memtables->flush();
+    } catch (...) {
+        ex = std::current_exception();
+    }
+    co_await _compaction_manager.remove(as_table_state());
+    co_await _sstable_deletion_gate.close();
+    co_await get_row_cache().invalidate(row_cache::external_updater([this] {
+        _main_sstables = _compaction_strategy.make_sstable_set(_schema);
+        _maintenance_sstables = make_maintenance_sstable_set();
+        _sstables = make_compound_sstable_set();
+    }));
+    _compaction_strategy.get_backlog_tracker().disable();
+    _cache.refresh_snapshot();
+
+    if (ex) {
+        std::rethrow_exception(std::move(ex));
```

Comment:
Why not `co_await coroutine::return_exception_ptr(std::move(ex))`?

Reply:
No reason, I just forgot this exists.

Reply:
BTW, we had some cascading finally causes before. How does that behave if multiple continuations in the chain throw? Will one exception be returned? Or will they be nested? The coroutinized version does the former.